Webpack4 TypeError: Cannot read property 'tap' of undefined

I think tap is new in webpak 4 "Replaced plugin calls with tap calls (new plugin system)"

```
➜  v2 git:(blue/webpack4) ✗ npm start

> mojo-marketplace-v2@2.0.0 start /Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2
> webpack --progress --watch --config webpack.config.babel.js

/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/uglifyjs-webpack-plugin/dist/index.js:267
            compilation.hooks.buildMobule.tap(plugin, buildMobuleFn);
                                          ^

TypeError: Cannot read property 'tap' of undefined
    at /Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/uglifyjs-webpack-plugin/dist/index.js:267:43
    at SyncHook.eval (eval at create (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/HookCodeFactory.js:17:12), <anonymous>:189:1)
    at SyncHook.lazyCompileHook [as _call] (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/Hook.js:35:21)
    at Compiler.newCompilation (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:407:26)
    at hooks.beforeCompile.callAsync.err (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:439:29)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/Hook.js:35:21)
    at Compiler.compile (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:434:28)
    at compiler.hooks.watchRun.callAsync.err (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:76:18)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/tapable/lib/Hook.js:35:21)
    at Watching._go (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:39:32)
    at Watching.compiler.readRecords.err (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:31:9)
    at Compiler.readRecords (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:331:11)
    at new Watching (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Watching.js:28:17)
    at Compiler.watch (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/lib/Compiler.js:161:10)
    at processOptions (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:508:14)
    at yargs.parse (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:516:3)
    at Object.parse (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/node_modules/yargs/yargs.js:539:18)
    at /Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:211:8
    at Object.<anonymous> (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack-cli/bin/webpack.js:518:3)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/aupeterson/Projects/ansible-playbooks/marketplace/git-repos/marketplace/Apps/v2/node_modules/webpack/bin/webpack.js:12:2)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Function.Module.runMain (module.js:701:10)
    at startup (bootstrap_node.js:190:16)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mojo-marketplace-v2@2.0.0 start: `webpack --progress --watch --config webpack.config.babel.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mojo-marketplace-v2@2.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/aupeterson/.npm/_logs/2018-02-24T00_29_16_700Z-debug.log
```

--- a/src/index.js
+++ b/src/index.js
@@ -1,8 +1,6 @@
-/*
-  MIT License http://www.opensource.org/licenses/mit-license.php
-  Author Tobias Koppers @sokra
+/* eslint-disable
+  no-param-reassign
 */
-
 import crypto from 'crypto';
 import { SourceMapConsumer } from 'source-map';
 import { SourceMapSource, RawSource, ConcatSource } from 'webpack-sources';
@@ -13,10 +11,6 @@ import serialize from 'serialize-javascript';
 import schema from './options.json';
 import Uglify from './uglify';
 import versions from './uglify/versions';
-
-/* eslint-disable
-  no-param-reassign
-*/
 
 const warningRegex = /\[.+:([0-9]+),([0-9]+)\]/;
 
@@ -95,7 +89,7 @@ class UglifyJsPlugin {
   apply(compiler) {
     const requestShortener = new RequestShortener(compiler.context);
 
-    const buildMobuleFn = (moduleArg) => {
+    const buildModuleFn = (moduleArg) => {
       // to get detailed location info about errors
       moduleArg.useSourceMap = true;
     };
@@ -105,8 +99,10 @@ class UglifyJsPlugin {
         cache: this.options.cache,
         parallel: this.options.parallel,
       });
+
       const uglifiedAssets = new WeakSet();
       const tasks = [];
+
       chunks.reduce((acc, chunk) => acc.concat(chunk.files || []), [])
         .concat(compilation.additionalChunkAssets || [])
         .filter(ModuleFilenameHelpers.matchObject.bind(null, this.options))
@@ -164,7 +160,14 @@ class UglifyJsPlugin {
 
             tasks.push(task);
           } catch (error) {
-            compilation.errors.push(UglifyJsPlugin.buildError(error, file, sourceMap, requestShortener));
+            compilation.errors.push(
+              UglifyJsPlugin.buildError(
+                error,
+                file,
+                sourceMap,
+                requestShortener,
+              ),
+            );
           }
         });
 
@@ -181,13 +184,25 @@ class UglifyJsPlugin {
           // Handling results
           // Error case: add errors, and go to next file
           if (error) {
-            compilation.errors.push(UglifyJsPlugin.buildError(error, file, sourceMap, requestShortener));
+            compilation.errors.push(
+              UglifyJsPlugin.buildError(
+                error,
+                file,
+                sourceMap,
+                requestShortener,
+              ),
+            );
+
             return;
           }
 
           let outputSource;
           if (map) {
-            outputSource = new SourceMapSource(code, file, JSON.parse(map), input, inputSourceMap);
+            outputSource = new SourceMapSource(
+              code,
+              file,
+              JSON.parse(map), input, inputSourceMap,
+            );
           } else {
             outputSource = new RawSource(code);
           }
@@ -197,9 +212,11 @@ class UglifyJsPlugin {
             // Add a banner to the original file
             if (this.options.extractComments.banner !== false) {
               let banner = this.options.extractComments.banner || `For license information please see ${commentsFile}`;
+
               if (typeof banner === 'function') {
                 banner = banner(commentsFile);
               }
+
               if (banner) {
                 outputSource = new ConcatSource(
                   `/*! ${banner} */\n`, outputSource,
@@ -208,6 +225,7 @@ class UglifyJsPlugin {
             }
 
             const commentsSource = new RawSource(`${extractedComments.join('\n\n')}\n`);
+
             if (commentsFile in compilation.assets) {
               // commentsFile already exists, append new comments...
               if (compilation.assets[commentsFile] instanceof ConcatSource) {
@@ -228,14 +246,24 @@ class UglifyJsPlugin {
 
           // Handling warnings
           if (warnings) {
-            const warnArr = UglifyJsPlugin.buildWarnings(warnings, file, sourceMap, this.options.warningsFilter, requestShortener);
+            const warnArr = UglifyJsPlugin.buildWarnings(
+              warnings,
+              file,
+              sourceMap,
+              this.options.warningsFilter,
+              requestShortener,
+            );
+
             if (warnArr.length > 0) {
-              compilation.warnings.push(new Error(`${file} from UglifyJs\n${warnArr.join('\n')}`));
+              compilation.warnings.push(
+                new Error(`${file} from UglifyJs\n${warnArr.join('\n')}`),
+              );
             }
           }
         });
 
         uglify.exit();
+
         callback();
       });
     };
@@ -245,7 +273,7 @@ class UglifyJsPlugin {
 
       compiler.hooks.compilation.tap(plugin, (compilation) => {
         if (this.options.sourceMap) {
-          compilation.hooks.buildMobule.tap(plugin, buildMobuleFn);
+          compilation.hooks.buildModule.tap(plugin, buildModuleFn);
         }
 
         compilation.hooks.optimizeChunkAssets.tapAsync(plugin, optimizeFn.bind(this, compilation));
@@ -253,7 +281,7 @@ class UglifyJsPlugin {
     } else {
       compiler.plugin('compilation', (compilation) => {
         if (this.options.sourceMap) {
-          compilation.plugin('build-module', buildMobuleFn);
+          compilation.plugin('build-module', buildModuleFn);
         }
 
         compilation.plugin('optimize-chunk-assets', optimizeFn.bind(this, compilation));

